### PR TITLE
Xiaoyu add editable container for info

### PR DESCRIPTION
**Pull request by 20chen-7**
# Description
Screenshot 2023-07-19 at 6 30 58 PM

## Related PRS (if any):
To test this backend PR you need to checkout the [#1025](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1025) frontend PR
…

## Main changes explained:
- Add information controller
- modifies routes
- add informationrouter
- modify server
- add information models
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
Login as Owner:
https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/b64a595e-2d8e-4e11-b645-30eec40ac6e8
Login as Admin:
https://github.com/OneCommunityGlobal/HGNRest/assets/102696148/60d049a5-5014-421a-aeb9-cd3add6a16b0
Login as Volunteer:
Screenshot 2023-07-19 at 6 21 02 PM

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Review by scoopstroop92 — COMMENTED** [review]
Left a detailed review in the frontend PR . I will approve the PR based on the feedback for my comments in the front end PR

**Comment by luisarevalo21** [comment]
Hello @20chen-7,

I reviewed your PR on the frontend and suggested some feedback. 
Overall your work on the PR was great. I'll approve this one once you review my suggestions and take them into consideration. 

Thanks

File: src/controllers/informationController.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+const informationController = function (Information) {
+  const getInformations = function (req, res) {
+    Information.find({}, 'infoName infoContent')
+      .then(results => res.status(200).send(results))
+      .catch(error => res.status(404).send(error));
+  };
+
+
+  const addInformation = function (req, res) {
```
Comment (iTvX) [inline src/controllers/informationController.js:15]: You first create a new Information and then create another new Information after checking for name uniqueness.  I suggest you may first check for name uniqueness, and then create the Information.

File: src/controllers/userProfileController.js [hunk 2]
```diff
@@ -763,6 +768,80 @@ const userProfileController = function (UserProfile) {
...
+      .then((users) => {
+        const infoCollections = users.map((user) => user.infoCollections);
+        res.status(200).json(infoCollections);
+      })
+      .catch((error) => {
+        res.status(500).send({ error: error.message });
+      });
+  };
+  
+  const putInfoCollections = function (req, res) {
```
Comment (iTvX) [inline src/controllers/userProfileController.js:782]: I noticed that you're using `UserProfile.updateMany({}, { $set: { infoCollections } })` to update the `infoCollections` field for all users. I would suggest considering including the user's ID and their corresponding `infoCollections` in the request body, and then iterating over these users and `infoCollections` in the method to update the `infoCollections` for each user. 

```javascript
const { userInfos } = req.body; // userInfos should be an array of { userId, infoCollections }
for (let userInfo of userInfos) {
  await UserProfile.updateOne({ _id: userInfo.userId }, { $set: { infoCollections: userInfo.infoCollections } });
}

**Review by DavidC0126 — CHANGES_REQUESTED** [review]
I have tested the PR with FE [PR#1025](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1025) and everything works as intended. I leave the comment on FE PR. 

However, when I pull down the branch and want to npm run start, it said that badgeController.js is missing so I manually add the file back to the BE (I did not see the file in this branch).

**Review by DavidC0126 — APPROVED** [review]
It looks good to me.

**Comment by 20chen-7** [comment]
> Hello @20chen-7,
> 
> I reviewed your PR on the frontend and suggested some feedback. Overall your work on the PR was great. I'll approve this one once you review my suggestions and take them into consideration.
> 
> Thanks

Fixed

**Review by ThapeloMasasa — COMMENTED** [review]
The functionality works as intended... added more information on the frontedn PR

**Review by veronicacheng2 — APPROVED** [review]
Hi @20chen-7, I have tested the PR together with #1052 and left comments there. Everything works as intended.

**Review by winghojackyli — APPROVED** [review]
I've tested this PR and it works great with frontend. Comment left on frontend.

**Review by TheDescendant39 — COMMENTED** [review]
Hi @20chen-7, I reviewed you PR and commented on [#1025](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1025#pullrequestreview-1575581126)

**Review by winghojackyli — APPROVED** [review]
Hi @20chen-7 , updated review is availble on the corresponding frontend PR. Comments left here: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1025#pullrequestreview-1581552404

**Pull request merged**